File: fibonacci.go
package main

import (
	"fmt"
	"time"
)

func main() {
	start := time.Now()

	result := fibonacci(30)
	fmt.Printf("Result: %d\n", result)

	elapsed := time.Since(start)
	fmt.Printf("Time: %v\n", elapsed)
}

func fibonacci(n int) int {
	if n < 2 {
		return n
	}
	return fibonacci(n-1) + fibonacci(n-2)
}

File: fibonacci.rs
use std::time::Instant;


fn fibonacci(n: i32) -> i32 {
    if n < 2 {
        return n;
    }
    fibonacci(n - 1) + fibonacci(n - 2)
}


fn main() {
    let start = Instant::now();

    let result = fibonacci(30);
    println!("Result: {}", result);

    let elapsed = start.elapsed();
    println!("Time: {:.3?}", elapsed);
}
